<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_35) on Mon Jun 01 10:50:28 EDT 2015 -->
<TITLE>
FastGenotype (HTSJDK API Documentation)
</TITLE>

<META NAME="date" CONTENT="2015-06-01">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="FastGenotype (HTSJDK API Documentation)";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/FastGenotype.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../index-all.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../htsjdk/variant/variantcontext/CommonInfo.html" title="class in htsjdk.variant.variantcontext"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../htsjdk/variant/variantcontext/Genotype.html" title="class in htsjdk.variant.variantcontext"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../index.html?htsjdk/variant/variantcontext/FastGenotype.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="FastGenotype.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;<A HREF="#fields_inherited_from_class_htsjdk.variant.variantcontext.Genotype">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>
<FONT SIZE="-1">
htsjdk.variant.variantcontext</FONT>
<BR>
Class FastGenotype</H2>
<PRE>
java.lang.Object
  <IMG SRC="../../../resources/inherit.gif" ALT="extended by "><A HREF="../../../htsjdk/variant/variantcontext/Genotype.html" title="class in htsjdk.variant.variantcontext">htsjdk.variant.variantcontext.Genotype</A>
      <IMG SRC="../../../resources/inherit.gif" ALT="extended by "><B>htsjdk.variant.variantcontext.FastGenotype</B>
</PRE>
<DL>
<DT><B>All Implemented Interfaces:</B> <DD>java.io.Serializable, java.lang.Comparable&lt;<A HREF="../../../htsjdk/variant/variantcontext/Genotype.html" title="class in htsjdk.variant.variantcontext">Genotype</A>&gt;</DD>
</DL>
<HR>
<DL>
<DT><PRE>public final class <B>FastGenotype</B><DT>extends <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html" title="class in htsjdk.variant.variantcontext">Genotype</A></DL>
</PRE>

<P>
This class encompasses all the basic information about a genotype.  It is immutable.

 A genotype has several key fields

 <ul>
 <li> a sample name, must be a non-null string</li>
 <li> an ordered list of alleles, intrepreted as the genotype of the sample,
    each allele for each chromosome given in order.  If alleles = [a*, t]
    then the sample is a/t, with a (the reference from the *) the first
    chromosome and t on the second chromosome</li>
 <li> an <code>isPhased</code> marker indicating where the alleles are phased with respect to some global
    coordinate system.  See VCF4.1 spec for a detailed discussion</li>
 <li> Inline, optimized <code>int</code>s and <code>int[]</code> values for:
 <ul>
      <li> GQ: the phred-scaled genotype quality, or <code>-1</code> if it's missing</li>
      <li> DP: the count of reads at this locus for this sample, or <code>-1</code> if missing</li>
      <li> AD: an array of counts of reads at this locus, one for each Allele at the site,
             that is, for each allele in the surrounding <code>VariantContext</code>.  <code>null</code> if missing.</li>
      <li> PL: phred-scaled genotype likelihoods in standard VCF4.1 order for
             all combinations of the alleles in the surrounding <code>VariantContext</code>, given
             the ploidy of the sample (from the alleles vector).  <code>null</code> if missing.</li>
 </ul>
 </li>

 <li> A general map from String keys to -&gt; Object values for all other attributes in
    this genotype.  Note that this map should not contain duplicate values for the
    standard bindings for GQ, DP, AD, and PL.  Genotype filters can be put into
    this genotype, but it isn't respected by the GATK in analyses</li>
</ul>

 <p>The only way to build a <code>Genotype</code> object is with a <code>GenotypeBuilder</code>, which permits values
 to be set in any order, which means that <code>GenotypeBuilder</code> may at some in the chain of
 sets pass through invalid states that are not permitted in a fully formed immutable
 <code>Genotype</code>.</p>

 <p>Note this is a simplified, refactored Genotype object based on the original
 generic (and slow) implementation from the original VariantContext + Genotype
 codebase.</p>
<P>

<P>
<DL>
<DT><B>Since:</B></DT>
  <DD>05/12</DD>
<DT><B>Author:</B></DT>
  <DD>Mark DePristo</DD>
<DT><B>See Also:</B><DD><A HREF="../../../serialized-form.html#htsjdk.variant.variantcontext.FastGenotype">Serialized Form</A></DL>
<HR>

<P>
<!-- =========== FIELD SUMMARY =========== -->

<A NAME="field_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Field Summary</B></FONT></TH>
</TR>
</TABLE>
&nbsp;<A NAME="fields_inherited_from_class_htsjdk.variant.variantcontext.Genotype"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Fields inherited from class htsjdk.variant.variantcontext.<A HREF="../../../htsjdk/variant/variantcontext/Genotype.html" title="class in htsjdk.variant.variantcontext">Genotype</A></B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#PHASED_ALLELE_SEPARATOR">PHASED_ALLELE_SEPARATOR</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#PRIMARY_KEYS">PRIMARY_KEYS</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#serialVersionUID">serialVersionUID</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#UNPHASED_ALLELE_SEPARATOR">UNPHASED_ALLELE_SEPARATOR</A></CODE></TD>
</TR>
</TABLE>
&nbsp;
<!-- ======== CONSTRUCTOR SUMMARY ======== -->

<A NAME="constructor_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Constructor Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected </CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/variant/variantcontext/FastGenotype.html#FastGenotype(java.lang.String, java.util.List, boolean, int, int, int[], int[], java.lang.String, java.util.Map)">FastGenotype</A></B>(java.lang.String&nbsp;sampleName,
             java.util.List&lt;<A HREF="../../../htsjdk/variant/variantcontext/Allele.html" title="class in htsjdk.variant.variantcontext">Allele</A>&gt;&nbsp;alleles,
             boolean&nbsp;isPhased,
             int&nbsp;GQ,
             int&nbsp;DP,
             int[]&nbsp;AD,
             int[]&nbsp;PL,
             java.lang.String&nbsp;filters,
             java.util.Map&lt;java.lang.String,java.lang.Object&gt;&nbsp;extendedAttributes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The only way to make one of these, for use by GenotypeBuilder only</TD>
</TR>
</TABLE>
&nbsp;
<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Method Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/variant/variantcontext/FastGenotype.html#getAD()">getAD</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../htsjdk/variant/variantcontext/Allele.html" title="class in htsjdk.variant.variantcontext">Allele</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/variant/variantcontext/FastGenotype.html#getAllele(int)">getAllele</A></B>(int&nbsp;i)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get the ith allele in this genotype</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../htsjdk/variant/variantcontext/Allele.html" title="class in htsjdk.variant.variantcontext">Allele</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/variant/variantcontext/FastGenotype.html#getAlleles()">getAlleles</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/variant/variantcontext/FastGenotype.html#getDP()">getDP</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.Map&lt;java.lang.String,java.lang.Object&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/variant/variantcontext/FastGenotype.html#getExtendedAttributes()">getExtendedAttributes</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the extended attributes for this object</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/variant/variantcontext/FastGenotype.html#getGQ()">getGQ</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns a phred-scaled quality score, or -1 if none is available</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/variant/variantcontext/FastGenotype.html#getPL()">getPL</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Unsafe low-level accessor the PL field itself, may be null.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/variant/variantcontext/FastGenotype.html#isPhased()">isPhased</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Are the alleles phased w.r.t.</TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_htsjdk.variant.variantcontext.Genotype"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class htsjdk.variant.variantcontext.<A HREF="../../../htsjdk/variant/variantcontext/Genotype.html" title="class in htsjdk.variant.variantcontext">Genotype</A></B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#compareTo(htsjdk.variant.variantcontext.Genotype)">compareTo</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#countAllele(htsjdk.variant.variantcontext.Allele)">countAllele</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#determineType()">determineType</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#getAlleleStrings()">getAlleleStrings</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#getAnyAttribute(java.lang.String)">getAnyAttribute</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#getAttributeAsDouble(java.lang.String, double)">getAttributeAsDouble</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#getAttributeAsInt(java.lang.String, int)">getAttributeAsInt</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#getAttributeAsString(java.lang.String, java.lang.String)">getAttributeAsString</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#getExtendedAttribute(java.lang.String)">getExtendedAttribute</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#getExtendedAttribute(java.lang.String, java.lang.Object)">getExtendedAttribute</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#getFilters()">getFilters</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#getGenotypeString()">getGenotypeString</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#getGenotypeString(boolean)">getGenotypeString</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#getLikelihoods()">getLikelihoods</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#getLikelihoodsString()">getLikelihoodsString</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#getLog10PError()">getLog10PError</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#getPhredScaledQual()">getPhredScaledQual</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#getPloidy()">getPloidy</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#getSampleName()">getSampleName</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#getType()">getType</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#hasAD()">hasAD</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#hasAnyAttribute(java.lang.String)">hasAnyAttribute</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#hasDP()">hasDP</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#hasExtendedAttribute(java.lang.String)">hasExtendedAttribute</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#hasForbiddenKey(java.util.Map)">hasForbiddenKey</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#hasGQ()">hasGQ</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#hasLikelihoods()">hasLikelihoods</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#hasLog10PError()">hasLog10PError</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#hasPL()">hasPL</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#isAvailable()">isAvailable</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#isCalled()">isCalled</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#isFiltered()">isFiltered</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#isForbiddenKey(java.lang.String)">isForbiddenKey</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#isHet()">isHet</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#isHetNonRef()">isHetNonRef</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#isHom()">isHom</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#isHomRef()">isHomRef</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#isHomVar()">isHomVar</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#isMixed()">isMixed</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#isNoCall()">isNoCall</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#isNonInformative()">isNonInformative</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#sameGenotype(htsjdk.variant.variantcontext.Genotype)">sameGenotype</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#sameGenotype(htsjdk.variant.variantcontext.Genotype, boolean)">sameGenotype</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#sortedString(java.util.Map)">sortedString</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#toBriefString()">toBriefString</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#toString()">toString</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#toStringIfExists(java.lang.String, int)">toStringIfExists</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#toStringIfExists(java.lang.String, int[])">toStringIfExists</A>, <A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#toStringIfExists(java.lang.String, java.lang.String)">toStringIfExists</A></CODE></TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_java.lang.Object"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class java.lang.Object</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>clone, equals, finalize, getClass, hashCode, notify, notifyAll, wait, wait, wait</CODE></TD>
</TR>
</TABLE>
&nbsp;
<P>

<!-- ========= CONSTRUCTOR DETAIL ======== -->

<A NAME="constructor_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Constructor Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="FastGenotype(java.lang.String, java.util.List, boolean, int, int, int[], int[], java.lang.String, java.util.Map)"><!-- --></A><H3>
FastGenotype</H3>
<PRE>
protected <B>FastGenotype</B>(java.lang.String&nbsp;sampleName,
                       java.util.List&lt;<A HREF="../../../htsjdk/variant/variantcontext/Allele.html" title="class in htsjdk.variant.variantcontext">Allele</A>&gt;&nbsp;alleles,
                       boolean&nbsp;isPhased,
                       int&nbsp;GQ,
                       int&nbsp;DP,
                       int[]&nbsp;AD,
                       int[]&nbsp;PL,
                       java.lang.String&nbsp;filters,
                       java.util.Map&lt;java.lang.String,java.lang.Object&gt;&nbsp;extendedAttributes)</PRE>
<DL>
<DD>The only way to make one of these, for use by GenotypeBuilder only
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>sampleName</CODE> - <DD><CODE>alleles</CODE> - <DD><CODE>isPhased</CODE> - <DD><CODE>GQ</CODE> - <DD><CODE>DP</CODE> - <DD><CODE>AD</CODE> - <DD><CODE>PL</CODE> - <DD><CODE>extendedAttributes</CODE> - </DL>
</DL>

<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Method Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="getAlleles()"><!-- --></A><H3>
getAlleles</H3>
<PRE>
public java.util.List&lt;<A HREF="../../../htsjdk/variant/variantcontext/Allele.html" title="class in htsjdk.variant.variantcontext">Allele</A>&gt; <B>getAlleles</B>()</PRE>
<DL>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#getAlleles()">getAlleles</A></CODE> in class <CODE><A HREF="../../../htsjdk/variant/variantcontext/Genotype.html" title="class in htsjdk.variant.variantcontext">Genotype</A></CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>the alleles for this genotype.  Cannot be null.  May be empty</DL>
</DD>
</DL>
<HR>

<A NAME="getAllele(int)"><!-- --></A><H3>
getAllele</H3>
<PRE>
public <A HREF="../../../htsjdk/variant/variantcontext/Allele.html" title="class in htsjdk.variant.variantcontext">Allele</A> <B>getAllele</B>(int&nbsp;i)</PRE>
<DL>
<DD><B>Description copied from class: <CODE><A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#getAllele(int)">Genotype</A></CODE></B></DD>
<DD>Get the ith allele in this genotype
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#getAllele(int)">getAllele</A></CODE> in class <CODE><A HREF="../../../htsjdk/variant/variantcontext/Genotype.html" title="class in htsjdk.variant.variantcontext">Genotype</A></CODE></DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>i</CODE> - the ith allele, must be &lt; the ploidy, starting with 0
<DT><B>Returns:</B><DD>the allele at position i, which cannot be null</DL>
</DD>
</DL>
<HR>

<A NAME="isPhased()"><!-- --></A><H3>
isPhased</H3>
<PRE>
public boolean <B>isPhased</B>()</PRE>
<DL>
<DD><B>Description copied from class: <CODE><A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#isPhased()">Genotype</A></CODE></B></DD>
<DD>Are the alleles phased w.r.t. the global phasing system?
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#isPhased()">isPhased</A></CODE> in class <CODE><A HREF="../../../htsjdk/variant/variantcontext/Genotype.html" title="class in htsjdk.variant.variantcontext">Genotype</A></CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>true if yes</DL>
</DD>
</DL>
<HR>

<A NAME="getDP()"><!-- --></A><H3>
getDP</H3>
<PRE>
public int <B>getDP</B>()</PRE>
<DL>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#getDP()">getDP</A></CODE> in class <CODE><A HREF="../../../htsjdk/variant/variantcontext/Genotype.html" title="class in htsjdk.variant.variantcontext">Genotype</A></CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>the sequencing depth of this sample, or -1 if this value is missing</DL>
</DD>
</DL>
<HR>

<A NAME="getAD()"><!-- --></A><H3>
getAD</H3>
<PRE>
public int[] <B>getAD</B>()</PRE>
<DL>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#getAD()">getAD</A></CODE> in class <CODE><A HREF="../../../htsjdk/variant/variantcontext/Genotype.html" title="class in htsjdk.variant.variantcontext">Genotype</A></CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>the count of reads, one for each allele in the surrounding Variant context,
      matching the corresponding allele, or null if this value is missing.  MUST
      NOT BE MODIFIED!</DL>
</DD>
</DL>
<HR>

<A NAME="getGQ()"><!-- --></A><H3>
getGQ</H3>
<PRE>
public int <B>getGQ</B>()</PRE>
<DL>
<DD><B>Description copied from class: <CODE><A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#getGQ()">Genotype</A></CODE></B></DD>
<DD>Returns a phred-scaled quality score, or -1 if none is available
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#getGQ()">getGQ</A></CODE> in class <CODE><A HREF="../../../htsjdk/variant/variantcontext/Genotype.html" title="class in htsjdk.variant.variantcontext">Genotype</A></CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getPL()"><!-- --></A><H3>
getPL</H3>
<PRE>
public int[] <B>getPL</B>()</PRE>
<DL>
<DD><B>Description copied from class: <CODE><A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#getPL()">Genotype</A></CODE></B></DD>
<DD>Unsafe low-level accessor the PL field itself, may be null.
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#getPL()">getPL</A></CODE> in class <CODE><A HREF="../../../htsjdk/variant/variantcontext/Genotype.html" title="class in htsjdk.variant.variantcontext">Genotype</A></CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>a pointer to the underlying PL data.  MUST NOT BE MODIFIED!</DL>
</DD>
</DL>
<HR>

<A NAME="getExtendedAttributes()"><!-- --></A><H3>
getExtendedAttributes</H3>
<PRE>
public java.util.Map&lt;java.lang.String,java.lang.Object&gt; <B>getExtendedAttributes</B>()</PRE>
<DL>
<DD><B>Description copied from class: <CODE><A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#getExtendedAttributes()">Genotype</A></CODE></B></DD>
<DD>Returns the extended attributes for this object
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../htsjdk/variant/variantcontext/Genotype.html#getExtendedAttributes()">getExtendedAttributes</A></CODE> in class <CODE><A HREF="../../../htsjdk/variant/variantcontext/Genotype.html" title="class in htsjdk.variant.variantcontext">Genotype</A></CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>is never null, but is often isEmpty()</DL>
</DD>
</DL>
<!-- ========= END OF CLASS DATA ========= -->
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/FastGenotype.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../index-all.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../htsjdk/variant/variantcontext/CommonInfo.html" title="class in htsjdk.variant.variantcontext"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../htsjdk/variant/variantcontext/Genotype.html" title="class in htsjdk.variant.variantcontext"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../index.html?htsjdk/variant/variantcontext/FastGenotype.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="FastGenotype.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;<A HREF="#fields_inherited_from_class_htsjdk.variant.variantcontext.Genotype">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
