<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_35) on Mon Feb 23 11:56:23 EST 2015 -->
<TITLE>
GenotypeLikelihoods (HTSJDK API Documentation)
</TITLE>

<META NAME="date" CONTENT="2015-02-23">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="GenotypeLikelihoods (HTSJDK API Documentation)";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/GenotypeLikelihoods.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../index-all.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../htsjdk/variant/variantcontext/GenotypeBuilder.html" title="class in htsjdk.variant.variantcontext"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.GenotypeLikelihoodsAllelePair.html" title="class in htsjdk.variant.variantcontext"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../index.html?htsjdk/variant/variantcontext/GenotypeLikelihoods.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="GenotypeLikelihoods.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;<A HREF="#nested_class_summary">NESTED</A>&nbsp;|&nbsp;<A HREF="#field_summary">FIELD</A>&nbsp;|&nbsp;CONSTR&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;<A HREF="#field_detail">FIELD</A>&nbsp;|&nbsp;CONSTR&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>
<FONT SIZE="-1">
htsjdk.variant.variantcontext</FONT>
<BR>
Class GenotypeLikelihoods</H2>
<PRE>
java.lang.Object
  <IMG SRC="../../../resources/inherit.gif" ALT="extended by "><B>htsjdk.variant.variantcontext.GenotypeLikelihoods</B>
</PRE>
<HR>
<DL>
<DT><PRE>public class <B>GenotypeLikelihoods</B><DT>extends java.lang.Object</DL>
</PRE>

<P>
<HR>

<P>
<!-- ======== NESTED CLASS SUMMARY ======== -->

<A NAME="nested_class_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Nested Class Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.GenotypeLikelihoodsAllelePair.html" title="class in htsjdk.variant.variantcontext">GenotypeLikelihoods.GenotypeLikelihoodsAllelePair</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;<!-- =========== FIELD SUMMARY =========== -->

<A NAME="field_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Field Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#MAX_ALT_ALLELES_THAT_CAN_BE_GENOTYPED">MAX_ALT_ALLELES_THAT_CAN_BE_GENOTYPED</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The maximum number of alleles that we can represent as genotype likelihoods</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#MAX_PL">MAX_PL</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected static&nbsp;int[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#PLindexConversion">PLindexConversion</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Method Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#calculatePLindex(int, int)">calculatePLindex</A></B>(int&nbsp;allele1Index,
                 int&nbsp;allele2Index)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#equals(java.lang.Object)">equals</A></B>(java.lang.Object&nbsp;aThat)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html" title="class in htsjdk.variant.variantcontext">GenotypeLikelihoods</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#fromGLField(java.lang.String)">fromGLField</A></B>(java.lang.String&nbsp;GLs)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<B>Deprecated.</B>&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html" title="class in htsjdk.variant.variantcontext">GenotypeLikelihoods</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#fromLog10Likelihoods(double[])">fromLog10Likelihoods</A></B>(double[]&nbsp;log10Likelihoods)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html" title="class in htsjdk.variant.variantcontext">GenotypeLikelihoods</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#fromPLField(java.lang.String)">fromPLField</A></B>(java.lang.String&nbsp;PLs)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html" title="class in htsjdk.variant.variantcontext">GenotypeLikelihoods</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#fromPLs(int[])">fromPLs</A></B>(int[]&nbsp;pls)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.GenotypeLikelihoodsAllelePair.html" title="class in htsjdk.variant.variantcontext">GenotypeLikelihoods.GenotypeLikelihoodsAllelePair</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#getAllelePair(int)">getAllelePair</A></B>(int&nbsp;PLindex)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;get the allele index pair for the given PL</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.GenotypeLikelihoodsAllelePair.html" title="class in htsjdk.variant.variantcontext">GenotypeLikelihoods.GenotypeLikelihoodsAllelePair</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#getAllelePairUsingDeprecatedOrdering(int)">getAllelePairUsingDeprecatedOrdering</A></B>(int&nbsp;PLindex)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<B>Deprecated.</B>&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.EnumMap&lt;<A HREF="../../../htsjdk/variant/variantcontext/GenotypeType.html" title="enum in htsjdk.variant.variantcontext">GenotypeType</A>,java.lang.Double&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#getAsMap(boolean)">getAsMap</A></B>(boolean&nbsp;normalizeFromLog10)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#getAsPLs()">getAsPLs</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#getAsString()">getAsString</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#getAsVector()">getAsVector</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the genotypes likelihoods in negative log10 vector format.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#getGQLog10FromLikelihoods(int, double[])">getGQLog10FromLikelihoods</A></B>(int&nbsp;iOfChoosenGenotype,
                          double[]&nbsp;likelihoods)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#getLog10GQ(htsjdk.variant.variantcontext.Genotype, java.util.List)">getLog10GQ</A></B>(<A HREF="../../../htsjdk/variant/variantcontext/Genotype.html" title="class in htsjdk.variant.variantcontext">Genotype</A>&nbsp;genotype,
           java.util.List&lt;<A HREF="../../../htsjdk/variant/variantcontext/Allele.html" title="class in htsjdk.variant.variantcontext">Allele</A>&gt;&nbsp;vcAlleles)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#getLog10GQ(htsjdk.variant.variantcontext.GenotypeType)">getLog10GQ</A></B>(<A HREF="../../../htsjdk/variant/variantcontext/GenotypeType.html" title="enum in htsjdk.variant.variantcontext">GenotypeType</A>&nbsp;genotype)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<B>Deprecated.</B>&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#getLog10GQ(htsjdk.variant.variantcontext.Genotype, htsjdk.variant.variantcontext.VariantContext)">getLog10GQ</A></B>(<A HREF="../../../htsjdk/variant/variantcontext/Genotype.html" title="class in htsjdk.variant.variantcontext">Genotype</A>&nbsp;genotype,
           <A HREF="../../../htsjdk/variant/variantcontext/VariantContext.html" title="class in htsjdk.variant.variantcontext">VariantContext</A>&nbsp;context)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#getPLIndecesOfAlleles(int, int)">getPLIndecesOfAlleles</A></B>(int&nbsp;allele1Index,
                      int&nbsp;allele2Index)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;get the PL indexes (AA, AB, BB) for the given allele pair; assumes allele1Index <= allele2Index.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#numLikelihoods(int, int)">numLikelihoods</A></B>(int&nbsp;numAlleles,
               int&nbsp;ploidy)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Compute how many likelihood elements are associated with the given number of alleles
 Equivalent to asking in how many ways N non-negative integers can add up to P is S(N,P)
 where P = ploidy (number of chromosomes) and N = total # of alleles.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#toString()">toString</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_java.lang.Object"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class java.lang.Object</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>clone, finalize, getClass, hashCode, notify, notifyAll, wait, wait, wait</CODE></TD>
</TR>
</TABLE>
&nbsp;
<P>

<!-- ============ FIELD DETAIL =========== -->

<A NAME="field_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Field Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="MAX_PL"><!-- --></A><H3>
MAX_PL</H3>
<PRE>
public static final int <B>MAX_PL</B></PRE>
<DL>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../constant-values.html#htsjdk.variant.variantcontext.GenotypeLikelihoods.MAX_PL">Constant Field Values</A></DL>
</DL>
<HR>

<A NAME="MAX_ALT_ALLELES_THAT_CAN_BE_GENOTYPED"><!-- --></A><H3>
MAX_ALT_ALLELES_THAT_CAN_BE_GENOTYPED</H3>
<PRE>
public static final int <B>MAX_ALT_ALLELES_THAT_CAN_BE_GENOTYPED</B></PRE>
<DL>
<DD>The maximum number of alleles that we can represent as genotype likelihoods
<P>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../constant-values.html#htsjdk.variant.variantcontext.GenotypeLikelihoods.MAX_ALT_ALLELES_THAT_CAN_BE_GENOTYPED">Constant Field Values</A></DL>
</DL>
<HR>

<A NAME="PLindexConversion"><!-- --></A><H3>
PLindexConversion</H3>
<PRE>
protected static final int[] <B>PLindexConversion</B></PRE>
<DL>
<DL>
</DL>
</DL>

<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Method Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="fromPLField(java.lang.String)"><!-- --></A><H3>
fromPLField</H3>
<PRE>
public static final <A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html" title="class in htsjdk.variant.variantcontext">GenotypeLikelihoods</A> <B>fromPLField</B>(java.lang.String&nbsp;PLs)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="fromGLField(java.lang.String)"><!-- --></A><H3>
fromGLField</H3>
<PRE>
<FONT SIZE="-1">@Deprecated
</FONT>public static final <A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html" title="class in htsjdk.variant.variantcontext">GenotypeLikelihoods</A> <B>fromGLField</B>(java.lang.String&nbsp;GLs)</PRE>
<DL>
<DD><B>Deprecated.</B>&nbsp;
<P>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="fromLog10Likelihoods(double[])"><!-- --></A><H3>
fromLog10Likelihoods</H3>
<PRE>
public static final <A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html" title="class in htsjdk.variant.variantcontext">GenotypeLikelihoods</A> <B>fromLog10Likelihoods</B>(double[]&nbsp;log10Likelihoods)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="fromPLs(int[])"><!-- --></A><H3>
fromPLs</H3>
<PRE>
public static final <A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html" title="class in htsjdk.variant.variantcontext">GenotypeLikelihoods</A> <B>fromPLs</B>(int[]&nbsp;pls)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getAsVector()"><!-- --></A><H3>
getAsVector</H3>
<PRE>
public double[] <B>getAsVector</B>()</PRE>
<DL>
<DD>Returns the genotypes likelihoods in negative log10 vector format.  pr{AA} = x, this
 vector returns math.log10(x) for each of the genotypes.  Can return null if the
 genotype likelihoods are "missing".
<P>
<DD><DL>

<DT><B>Returns:</B><DD></DL>
</DD>
</DL>
<HR>

<A NAME="getAsPLs()"><!-- --></A><H3>
getAsPLs</H3>
<PRE>
public int[] <B>getAsPLs</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="toString()"><!-- --></A><H3>
toString</H3>
<PRE>
public java.lang.String <B>toString</B>()</PRE>
<DL>
<DD><DL>
<DT><B>Overrides:</B><DD><CODE>toString</CODE> in class <CODE>java.lang.Object</CODE></DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getAsString()"><!-- --></A><H3>
getAsString</H3>
<PRE>
public java.lang.String <B>getAsString</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="equals(java.lang.Object)"><!-- --></A><H3>
equals</H3>
<PRE>
public boolean <B>equals</B>(java.lang.Object&nbsp;aThat)</PRE>
<DL>
<DD><DL>
<DT><B>Overrides:</B><DD><CODE>equals</CODE> in class <CODE>java.lang.Object</CODE></DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getAsMap(boolean)"><!-- --></A><H3>
getAsMap</H3>
<PRE>
public java.util.EnumMap&lt;<A HREF="../../../htsjdk/variant/variantcontext/GenotypeType.html" title="enum in htsjdk.variant.variantcontext">GenotypeType</A>,java.lang.Double&gt; <B>getAsMap</B>(boolean&nbsp;normalizeFromLog10)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getLog10GQ(htsjdk.variant.variantcontext.GenotypeType)"><!-- --></A><H3>
getLog10GQ</H3>
<PRE>
<FONT SIZE="-1">@Deprecated
</FONT>public double <B>getLog10GQ</B>(<A HREF="../../../htsjdk/variant/variantcontext/GenotypeType.html" title="enum in htsjdk.variant.variantcontext">GenotypeType</A>&nbsp;genotype)</PRE>
<DL>
<DD><B>Deprecated.</B>&nbsp;
<P>
<DD>This is really dangerous and returns completely wrong results for genotypes from a multi-allelic context.
 Use getLog10GQ(Genotype,VariantContext) or getLog10GQ(Genotype,List<Allele>) in place of it.

 If you **know** you're biallelic, use getGQLog10FromLikelihoods directly.
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>genotype</CODE> - - actually a genotype type (no call, hom ref, het, hom var)
<DT><B>Returns:</B><DD>an unsafe quantity that could be negative. In the bi-allelic case, the GQ resulting from best minus next best (if the type is the best).</DL>
</DD>
</DL>
<HR>

<A NAME="getLog10GQ(htsjdk.variant.variantcontext.Genotype, java.util.List)"><!-- --></A><H3>
getLog10GQ</H3>
<PRE>
public double <B>getLog10GQ</B>(<A HREF="../../../htsjdk/variant/variantcontext/Genotype.html" title="class in htsjdk.variant.variantcontext">Genotype</A>&nbsp;genotype,
                         java.util.List&lt;<A HREF="../../../htsjdk/variant/variantcontext/Allele.html" title="class in htsjdk.variant.variantcontext">Allele</A>&gt;&nbsp;vcAlleles)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getLog10GQ(htsjdk.variant.variantcontext.Genotype, htsjdk.variant.variantcontext.VariantContext)"><!-- --></A><H3>
getLog10GQ</H3>
<PRE>
public double <B>getLog10GQ</B>(<A HREF="../../../htsjdk/variant/variantcontext/Genotype.html" title="class in htsjdk.variant.variantcontext">Genotype</A>&nbsp;genotype,
                         <A HREF="../../../htsjdk/variant/variantcontext/VariantContext.html" title="class in htsjdk.variant.variantcontext">VariantContext</A>&nbsp;context)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getGQLog10FromLikelihoods(int, double[])"><!-- --></A><H3>
getGQLog10FromLikelihoods</H3>
<PRE>
public static double <B>getGQLog10FromLikelihoods</B>(int&nbsp;iOfChoosenGenotype,
                                               double[]&nbsp;likelihoods)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="numLikelihoods(int, int)"><!-- --></A><H3>
numLikelihoods</H3>
<PRE>
public static int <B>numLikelihoods</B>(int&nbsp;numAlleles,
                                 int&nbsp;ploidy)</PRE>
<DL>
<DD>Compute how many likelihood elements are associated with the given number of alleles
 Equivalent to asking in how many ways N non-negative integers can add up to P is S(N,P)
 where P = ploidy (number of chromosomes) and N = total # of alleles.
 Each chromosome can be in one single state (0,...,N-1) and there are P of them.
 Naive solution would be to store N*P likelihoods, but this is not necessary because we can't distinguish chromosome states, but rather
 only total number of alt allele counts in all chromosomes.

 For example, S(3,2) = 6: For alleles A,B,C, on a diploid organism we have six possible genotypes:
 AA,AB,BB,AC,BC,CC.
 Another way of expressing is with vector (#of A alleles, # of B alleles, # of C alleles)
 which is then, for ordering above, (2,0,0), (1,1,0), (0,2,0), (1,1,0), (0,1,1), (0,0,2)
 In general, for P=2 (regular biallelic), then S(N,2) = N*(N+1)/2

 Note this method caches the value for most common num Allele / ploidy combinations for efficiency

 Recursive implementation:
   S(N,P) = sum_{k=0}^P S(N-1,P-k)
  because if we have N integers, we can condition 1 integer to be = k, and then N-1 integers have to sum to P-K
 With initial conditions
   S(N,1) = N  (only way to have N integers add up to 1 is all-zeros except one element with a one. There are N of these vectors)
   S(1,P) = 1 (only way to have 1 integer add to P is with that integer P itself).
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>numAlleles</CODE> - Number of alleles (including ref)<DD><CODE>ploidy</CODE> - Ploidy, or number of chromosomes in set
<DT><B>Returns:</B><DD>Number of likelihood elements we need to hold.</DL>
</DD>
</DL>
<HR>

<A NAME="calculatePLindex(int, int)"><!-- --></A><H3>
calculatePLindex</H3>
<PRE>
public static int <B>calculatePLindex</B>(int&nbsp;allele1Index,
                                   int&nbsp;allele2Index)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getAllelePair(int)"><!-- --></A><H3>
getAllelePair</H3>
<PRE>
public static <A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.GenotypeLikelihoodsAllelePair.html" title="class in htsjdk.variant.variantcontext">GenotypeLikelihoods.GenotypeLikelihoodsAllelePair</A> <B>getAllelePair</B>(int&nbsp;PLindex)</PRE>
<DL>
<DD>get the allele index pair for the given PL
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>PLindex</CODE> - the PL index
<DT><B>Returns:</B><DD>the allele index pair</DL>
</DD>
</DL>
<HR>

<A NAME="getAllelePairUsingDeprecatedOrdering(int)"><!-- --></A><H3>
getAllelePairUsingDeprecatedOrdering</H3>
<PRE>
<FONT SIZE="-1">@Deprecated
</FONT>public static <A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.GenotypeLikelihoodsAllelePair.html" title="class in htsjdk.variant.variantcontext">GenotypeLikelihoods.GenotypeLikelihoodsAllelePair</A> <B>getAllelePairUsingDeprecatedOrdering</B>(int&nbsp;PLindex)</PRE>
<DL>
<DD><B>Deprecated.</B>&nbsp;
<P>
<DD>get the allele index pair for the given PL using the deprecated PL ordering:
    AA,AB,AC,AD,BB,BC,BD,CC,CD,DD instead of AA,AB,BB,AC,BC,CC,AD,BD,CD,DD.
 Although it's painful to keep this conversion around, our DiploidSNPGenotypeLikelihoods class uses the deprecated
    ordering and I know with certainty that external users have built code on top of it; changing it now would
    cause a whole lot of heartache for our collaborators, so for now at least there's a standard conversion method.
 This method assumes at most 3 alternate alleles.
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>PLindex</CODE> - the PL index
<DT><B>Returns:</B><DD>the allele index pair</DL>
</DD>
</DL>
<HR>

<A NAME="getPLIndecesOfAlleles(int, int)"><!-- --></A><H3>
getPLIndecesOfAlleles</H3>
<PRE>
public static int[] <B>getPLIndecesOfAlleles</B>(int&nbsp;allele1Index,
                                          int&nbsp;allele2Index)</PRE>
<DL>
<DD>get the PL indexes (AA, AB, BB) for the given allele pair; assumes allele1Index <= allele2Index.
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>allele1Index</CODE> - the index in VariantContext.getAllele() of the first allele<DD><CODE>allele2Index</CODE> - the index in VariantContext.getAllele() of the second allele
<DT><B>Returns:</B><DD>the PL indexes</DL>
</DD>
</DL>
<!-- ========= END OF CLASS DATA ========= -->
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/GenotypeLikelihoods.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../index-all.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../htsjdk/variant/variantcontext/GenotypeBuilder.html" title="class in htsjdk.variant.variantcontext"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.GenotypeLikelihoodsAllelePair.html" title="class in htsjdk.variant.variantcontext"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../index.html?htsjdk/variant/variantcontext/GenotypeLikelihoods.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="GenotypeLikelihoods.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;<A HREF="#nested_class_summary">NESTED</A>&nbsp;|&nbsp;<A HREF="#field_summary">FIELD</A>&nbsp;|&nbsp;CONSTR&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;<A HREF="#field_detail">FIELD</A>&nbsp;|&nbsp;CONSTR&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
