<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_65) on Mon Dec 01 10:26:52 EST 2014 -->
<TITLE>
Uses of Package htsjdk.variant.variantcontext (HTSJDK API Documentation)
</TITLE>

<META NAME="date" CONTENT="2014-12-01">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Uses of Package htsjdk.variant.variantcontext (HTSJDK API Documentation)";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Use</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../index-all.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV&nbsp;
&nbsp;NEXT</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../index.html?htsjdk/variant/variantcontext/package-use.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="package-use.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<CENTER>
<H2>
<B>Uses of Package<br>htsjdk.variant.variantcontext</B></H2>
</CENTER>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Packages that use <A HREF="../../../htsjdk/variant/variantcontext/package-summary.html">htsjdk.variant.variantcontext</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#htsjdk.variant.bcf2"><B>htsjdk.variant.bcf2</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#htsjdk.variant.variantcontext"><B>htsjdk.variant.variantcontext</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#htsjdk.variant.variantcontext.writer"><B>htsjdk.variant.variantcontext.writer</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#htsjdk.variant.vcf"><B>htsjdk.variant.vcf</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="htsjdk.variant.bcf2"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../htsjdk/variant/variantcontext/package-summary.html">htsjdk.variant.variantcontext</A> used by <A HREF="../../../htsjdk/variant/bcf2/package-summary.html">htsjdk.variant.bcf2</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../htsjdk/variant/variantcontext/class-use/Allele.html#htsjdk.variant.bcf2"><B>Allele</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Immutable representation of an allele

 Types of alleles:

 Ref: a t C g a // C is the reference base

    : a t G g a // C base is a G in some individuals

    : a t - g a // C base is deleted w.r.t.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../htsjdk/variant/variantcontext/class-use/GenotypeBuilder.html#htsjdk.variant.bcf2"><B>GenotypeBuilder</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A builder class for genotypes

 Provides convenience setter methods for all of the Genotype field
 values.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../htsjdk/variant/variantcontext/class-use/LazyGenotypesContext.LazyData.html#htsjdk.variant.bcf2"><B>LazyGenotypesContext.LazyData</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the data used in the full GenotypesContext constructor

 <A HREF="../../../htsjdk/variant/variantcontext/GenotypesContext.html#GenotypesContext(java.util.ArrayList, java.util.Map, java.util.List)"><CODE>GenotypesContext.GenotypesContext(java.util.ArrayList, java.util.Map, java.util.List)</CODE></A></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../htsjdk/variant/variantcontext/class-use/LazyGenotypesContext.LazyParser.html#htsjdk.variant.bcf2"><B>LazyGenotypesContext.LazyParser</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Simple lazy parser interface.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../htsjdk/variant/variantcontext/class-use/VariantContext.html#htsjdk.variant.bcf2"><B>VariantContext</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Class VariantContext

 == High-level overview ==

 The VariantContext object is a single general class system for representing genetic variation data composed of:

 * Allele: representing single genetic haplotypes (A, T, ATC, -) (note that null alleles are used here for illustration; see the Allele class for how to represent indels)
 * Genotype: an assignment of alleles for each chromosome of a single named sample at a particular locus
 * VariantContext: an abstract class holding all segregating alleles at a locus as well as genotypes
    for multiple individuals containing alleles at that locus

 The class system works by defining segregating alleles, creating a variant context representing the segregating
 information at a locus, and potentially creating and associating genotypes with individuals in the context.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="htsjdk.variant.variantcontext"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../htsjdk/variant/variantcontext/package-summary.html">htsjdk.variant.variantcontext</A> used by <A HREF="../../../htsjdk/variant/variantcontext/package-summary.html">htsjdk.variant.variantcontext</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../htsjdk/variant/variantcontext/class-use/Allele.html#htsjdk.variant.variantcontext"><B>Allele</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Immutable representation of an allele

 Types of alleles:

 Ref: a t C g a // C is the reference base

    : a t G g a // C base is a G in some individuals

    : a t - g a // C base is deleted w.r.t.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../htsjdk/variant/variantcontext/class-use/CommonInfo.html#htsjdk.variant.variantcontext"><B>CommonInfo</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Common utility routines for VariantContext and Genotype</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../htsjdk/variant/variantcontext/class-use/Genotype.html#htsjdk.variant.variantcontext"><B>Genotype</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This class encompasses all the basic information about a genotype.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../htsjdk/variant/variantcontext/class-use/GenotypeBuilder.html#htsjdk.variant.variantcontext"><B>GenotypeBuilder</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A builder class for genotypes

 Provides convenience setter methods for all of the Genotype field
 values.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../htsjdk/variant/variantcontext/class-use/GenotypeLikelihoods.html#htsjdk.variant.variantcontext"><B>GenotypeLikelihoods</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../htsjdk/variant/variantcontext/class-use/GenotypeLikelihoods.GenotypeLikelihoodsAllelePair.html#htsjdk.variant.variantcontext"><B>GenotypeLikelihoods.GenotypeLikelihoodsAllelePair</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../htsjdk/variant/variantcontext/class-use/GenotypesContext.html#htsjdk.variant.variantcontext"><B>GenotypesContext</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Represents an ordered collection of Genotype objects</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../htsjdk/variant/variantcontext/class-use/GenotypeType.html#htsjdk.variant.variantcontext"><B>GenotypeType</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Summary types for Genotype objects</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../htsjdk/variant/variantcontext/class-use/LazyGenotypesContext.LazyData.html#htsjdk.variant.variantcontext"><B>LazyGenotypesContext.LazyData</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the data used in the full GenotypesContext constructor

 <A HREF="../../../htsjdk/variant/variantcontext/GenotypesContext.html#GenotypesContext(java.util.ArrayList, java.util.Map, java.util.List)"><CODE>GenotypesContext.GenotypesContext(java.util.ArrayList, java.util.Map, java.util.List)</CODE></A></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../htsjdk/variant/variantcontext/class-use/LazyGenotypesContext.LazyParser.html#htsjdk.variant.variantcontext"><B>LazyGenotypesContext.LazyParser</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Simple lazy parser interface.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../htsjdk/variant/variantcontext/class-use/VariantContext.html#htsjdk.variant.variantcontext"><B>VariantContext</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Class VariantContext

 == High-level overview ==

 The VariantContext object is a single general class system for representing genetic variation data composed of:

 * Allele: representing single genetic haplotypes (A, T, ATC, -) (note that null alleles are used here for illustration; see the Allele class for how to represent indels)
 * Genotype: an assignment of alleles for each chromosome of a single named sample at a particular locus
 * VariantContext: an abstract class holding all segregating alleles at a locus as well as genotypes
    for multiple individuals containing alleles at that locus

 The class system works by defining segregating alleles, creating a variant context representing the segregating
 information at a locus, and potentially creating and associating genotypes with individuals in the context.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../htsjdk/variant/variantcontext/class-use/VariantContext.Type.html#htsjdk.variant.variantcontext"><B>VariantContext.Type</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;see: http://www.ncbi.nlm.nih.gov/bookshelf/br.fcgi?book=handbook&part=ch5&rendertype=table&id=ch5.ch5_t3

 Format:
 dbSNP variation class
 Rules for assigning allele classes
 Sample allele definition

 Single Nucleotide Polymorphisms (SNPs)a
   Strictly defined as single base substitutions involving A, T, C, or G.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../htsjdk/variant/variantcontext/class-use/VariantContext.Validation.html#htsjdk.variant.variantcontext"><B>VariantContext.Validation</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../htsjdk/variant/variantcontext/class-use/VariantContextBuilder.html#htsjdk.variant.variantcontext"><B>VariantContextBuilder</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Builder class for VariantContext

 Some basic assumptions here:

 1 -- data isn't protectively copied.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../htsjdk/variant/variantcontext/class-use/VariantContextUtils.JexlVCMatchExp.html#htsjdk.variant.variantcontext"><B>VariantContextUtils.JexlVCMatchExp</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A simple but common wrapper for matching VariantContext objects using JEXL expressions</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="htsjdk.variant.variantcontext.writer"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../htsjdk/variant/variantcontext/package-summary.html">htsjdk.variant.variantcontext</A> used by <A HREF="../../../htsjdk/variant/variantcontext/writer/package-summary.html">htsjdk.variant.variantcontext.writer</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../htsjdk/variant/variantcontext/class-use/Genotype.html#htsjdk.variant.variantcontext.writer"><B>Genotype</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This class encompasses all the basic information about a genotype.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../htsjdk/variant/variantcontext/class-use/VariantContext.html#htsjdk.variant.variantcontext.writer"><B>VariantContext</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Class VariantContext

 == High-level overview ==

 The VariantContext object is a single general class system for representing genetic variation data composed of:

 * Allele: representing single genetic haplotypes (A, T, ATC, -) (note that null alleles are used here for illustration; see the Allele class for how to represent indels)
 * Genotype: an assignment of alleles for each chromosome of a single named sample at a particular locus
 * VariantContext: an abstract class holding all segregating alleles at a locus as well as genotypes
    for multiple individuals containing alleles at that locus

 The class system works by defining segregating alleles, creating a variant context representing the segregating
 information at a locus, and potentially creating and associating genotypes with individuals in the context.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="htsjdk.variant.vcf"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Classes in <A HREF="../../../htsjdk/variant/variantcontext/package-summary.html">htsjdk.variant.variantcontext</A> used by <A HREF="../../../htsjdk/variant/vcf/package-summary.html">htsjdk.variant.vcf</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../htsjdk/variant/variantcontext/class-use/Allele.html#htsjdk.variant.vcf"><B>Allele</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Immutable representation of an allele

 Types of alleles:

 Ref: a t C g a // C is the reference base

    : a t G g a // C base is a G in some individuals

    : a t - g a // C base is deleted w.r.t.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../htsjdk/variant/variantcontext/class-use/LazyGenotypesContext.LazyData.html#htsjdk.variant.vcf"><B>LazyGenotypesContext.LazyData</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the data used in the full GenotypesContext constructor

 <A HREF="../../../htsjdk/variant/variantcontext/GenotypesContext.html#GenotypesContext(java.util.ArrayList, java.util.Map, java.util.List)"><CODE>GenotypesContext.GenotypesContext(java.util.ArrayList, java.util.Map, java.util.List)</CODE></A></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../htsjdk/variant/variantcontext/class-use/VariantContext.html#htsjdk.variant.vcf"><B>VariantContext</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Class VariantContext

 == High-level overview ==

 The VariantContext object is a single general class system for representing genetic variation data composed of:

 * Allele: representing single genetic haplotypes (A, T, ATC, -) (note that null alleles are used here for illustration; see the Allele class for how to represent indels)
 * Genotype: an assignment of alleles for each chromosome of a single named sample at a particular locus
 * VariantContext: an abstract class holding all segregating alleles at a locus as well as genotypes
    for multiple individuals containing alleles at that locus

 The class system works by defining segregating alleles, creating a variant context representing the segregating
 information at a locus, and potentially creating and associating genotypes with individuals in the context.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><B><A HREF="../../../htsjdk/variant/variantcontext/class-use/VariantContextComparator.html#htsjdk.variant.vcf"><B>VariantContextComparator</B></A></B>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A Comparator that orders VariantContexts by the ordering of the contigs/chromosomes in the List
 provided at construction time, then by start position with each contig/chromosome.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Use</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../index-all.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV&nbsp;
&nbsp;NEXT</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../index.html?htsjdk/variant/variantcontext/package-use.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="package-use.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
