<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_35) on Wed Oct 07 15:06:35 EDT 2015 -->
<TITLE>
SequenceUtil (HTSJDK API Documentation)
</TITLE>

<META NAME="date" CONTENT="2015-10-07">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="SequenceUtil (HTSJDK API Documentation)";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/SequenceUtil.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../index-all.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../htsjdk/samtools/util/SamRecordWithOrdinal.html" title="class in htsjdk.samtools.util"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../htsjdk/samtools/util/SequenceUtil.SequenceListsDifferException.html" title="class in htsjdk.samtools.util"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../index.html?htsjdk/samtools/util/SequenceUtil.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="SequenceUtil.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;<A HREF="#nested_class_summary">NESTED</A>&nbsp;|&nbsp;<A HREF="#field_summary">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;<A HREF="#field_detail">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>
<FONT SIZE="-1">
htsjdk.samtools.util</FONT>
<BR>
Class SequenceUtil</H2>
<PRE>
java.lang.Object
  <IMG SRC="../../../resources/inherit.gif" ALT="extended by "><B>htsjdk.samtools.util.SequenceUtil</B>
</PRE>
<HR>
<DL>
<DT><PRE>public class <B>SequenceUtil</B><DT>extends java.lang.Object</DL>
</PRE>

<P>
<HR>

<P>
<!-- ======== NESTED CLASS SUMMARY ======== -->

<A NAME="nested_class_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Nested Class Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.SequenceListsDifferException.html" title="class in htsjdk.samtools.util">SequenceUtil.SequenceListsDifferException</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;<!-- =========== FIELD SUMMARY =========== -->

<A NAME="field_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Field Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;byte</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#a">a</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Byte typed variables for all normal bases.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;byte</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#A">A</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Byte typed variables for all normal bases.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;byte</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#c">c</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Byte typed variables for all normal bases.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;byte</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#C">C</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Byte typed variables for all normal bases.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;byte</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#g">g</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Byte typed variables for all normal bases.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;byte</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#G">G</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Byte typed variables for all normal bases.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;byte</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#n">n</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Byte typed variables for all normal bases.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;byte</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#N">N</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Byte typed variables for all normal bases.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;byte</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#t">t</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Byte typed variables for all normal bases.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;byte</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#T">T</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Byte typed variables for all normal bases.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;byte[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#VALID_BASES_LOWER">VALID_BASES_LOWER</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;byte[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#VALID_BASES_UPPER">VALID_BASES_UPPER</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<!-- ======== CONSTRUCTOR SUMMARY ======== -->

<A NAME="constructor_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Constructor Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#SequenceUtil()">SequenceUtil</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Method Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#areSequenceDictionariesEqual(htsjdk.samtools.SAMSequenceDictionary, htsjdk.samtools.SAMSequenceDictionary)">areSequenceDictionariesEqual</A></B>(<A HREF="../../../htsjdk/samtools/SAMSequenceDictionary.html" title="class in htsjdk.samtools">SAMSequenceDictionary</A>&nbsp;s1,
                             <A HREF="../../../htsjdk/samtools/SAMSequenceDictionary.html" title="class in htsjdk.samtools">SAMSequenceDictionary</A>&nbsp;s2)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns true if both parameters are null or equal, otherwise returns false</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#assertSequenceDictionariesEqual(htsjdk.samtools.SAMSequenceDictionary, htsjdk.samtools.SAMSequenceDictionary)">assertSequenceDictionariesEqual</A></B>(<A HREF="../../../htsjdk/samtools/SAMSequenceDictionary.html" title="class in htsjdk.samtools">SAMSequenceDictionary</A>&nbsp;s1,
                                <A HREF="../../../htsjdk/samtools/SAMSequenceDictionary.html" title="class in htsjdk.samtools">SAMSequenceDictionary</A>&nbsp;s2)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Throws an exception if both parameters are non-null and unequal.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#assertSequenceDictionariesEqual(htsjdk.samtools.SAMSequenceDictionary, htsjdk.samtools.SAMSequenceDictionary, java.io.File, java.io.File)">assertSequenceDictionariesEqual</A></B>(<A HREF="../../../htsjdk/samtools/SAMSequenceDictionary.html" title="class in htsjdk.samtools">SAMSequenceDictionary</A>&nbsp;s1,
                                <A HREF="../../../htsjdk/samtools/SAMSequenceDictionary.html" title="class in htsjdk.samtools">SAMSequenceDictionary</A>&nbsp;s2,
                                java.io.File&nbsp;f1,
                                java.io.File&nbsp;f2)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Throws an exception if both parameters are non-null and unequal, including the filenames.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#assertSequenceListsEqual(java.util.List, java.util.List)">assertSequenceListsEqual</A></B>(java.util.List&lt;<A HREF="../../../htsjdk/samtools/SAMSequenceRecord.html" title="class in htsjdk.samtools">SAMSequenceRecord</A>&gt;&nbsp;s1,
                         java.util.List&lt;<A HREF="../../../htsjdk/samtools/SAMSequenceRecord.html" title="class in htsjdk.samtools">SAMSequenceRecord</A>&gt;&nbsp;s2)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Throws an exception only if both parameters are not null</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#basesEqual(byte, byte)">basesEqual</A></B>(byte&nbsp;lhs,
           byte&nbsp;rhs)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Attempts to efficiently compare two bases stored as bytes for equality.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#bisulfiteBasesEqual(boolean, byte, byte)">bisulfiteBasesEqual</A></B>(boolean&nbsp;negativeStrand,
                    byte&nbsp;read,
                    byte&nbsp;reference)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns true if the bases are equal OR if the mismatch cannot be accounted for by
 bisfulite treatment.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#bisulfiteBasesEqual(byte, byte)">bisulfiteBasesEqual</A></B>(byte&nbsp;read,
                    byte&nbsp;reference)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#calculateGc(byte[])">calculateGc</A></B>(byte[]&nbsp;bases)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Calculates the fraction of bases that are G/C in the sequence.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;byte[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#calculateMD5(byte[], int, int)">calculateMD5</A></B>(byte[]&nbsp;data,
             int&nbsp;offset,
             int&nbsp;len)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#calculateMD5String(byte[])">calculateMD5String</A></B>(byte[]&nbsp;data)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#calculateMD5String(byte[], int, int)">calculateMD5String</A></B>(byte[]&nbsp;data,
                   int&nbsp;offset,
                   int&nbsp;len)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#calculateMdAndNmTags(htsjdk.samtools.SAMRecord, byte[], boolean, boolean)">calculateMdAndNmTags</A></B>(<A HREF="../../../htsjdk/samtools/SAMRecord.html" title="class in htsjdk.samtools">SAMRecord</A>&nbsp;record,
                     byte[]&nbsp;ref,
                     boolean&nbsp;calcMD,
                     boolean&nbsp;calcNM)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A rip off samtools bam_md.c</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#calculateSamNmTag(htsjdk.samtools.SAMRecord, byte[])">calculateSamNmTag</A></B>(<A HREF="../../../htsjdk/samtools/SAMRecord.html" title="class in htsjdk.samtools">SAMRecord</A>&nbsp;read,
                  byte[]&nbsp;referenceBases)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Calculates the for the predefined NM tag from the SAM spec.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#calculateSamNmTag(htsjdk.samtools.SAMRecord, byte[], int)">calculateSamNmTag</A></B>(<A HREF="../../../htsjdk/samtools/SAMRecord.html" title="class in htsjdk.samtools">SAMRecord</A>&nbsp;read,
                  byte[]&nbsp;referenceBases,
                  int&nbsp;referenceOffset)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Calculates the for the predefined NM tag from the SAM spec.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#calculateSamNmTag(htsjdk.samtools.SAMRecord, byte[], int, boolean)">calculateSamNmTag</A></B>(<A HREF="../../../htsjdk/samtools/SAMRecord.html" title="class in htsjdk.samtools">SAMRecord</A>&nbsp;read,
                  byte[]&nbsp;referenceBases,
                  int&nbsp;referenceOffset,
                  boolean&nbsp;bisulfiteSequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Calculates the for the predefined NM tag from the SAM spec.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#calculateSamNmTag(htsjdk.samtools.SAMRecord, char[], int)">calculateSamNmTag</A></B>(<A HREF="../../../htsjdk/samtools/SAMRecord.html" title="class in htsjdk.samtools">SAMRecord</A>&nbsp;read,
                  char[]&nbsp;referenceBases,
                  int&nbsp;referenceOffset)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<B>Deprecated.</B>&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#calculateSamNmTagFromCigar(htsjdk.samtools.SAMRecord)">calculateSamNmTagFromCigar</A></B>(<A HREF="../../../htsjdk/samtools/SAMRecord.html" title="class in htsjdk.samtools">SAMRecord</A>&nbsp;record)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Attempts to calculate the for the predefined NM tag from the SAM spec using the cigar string alone.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;byte</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#complement(byte)">complement</A></B>(byte&nbsp;b)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the complement of a single byte.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#countDeletedBases(htsjdk.samtools.Cigar)">countDeletedBases</A></B>(<A HREF="../../../htsjdk/samtools/Cigar.html" title="class in htsjdk.samtools">Cigar</A>&nbsp;cigar)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#countDeletedBases(htsjdk.samtools.SAMRecord)">countDeletedBases</A></B>(<A HREF="../../../htsjdk/samtools/SAMRecord.html" title="class in htsjdk.samtools">SAMRecord</A>&nbsp;read)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#countInsertedBases(htsjdk.samtools.Cigar)">countInsertedBases</A></B>(<A HREF="../../../htsjdk/samtools/Cigar.html" title="class in htsjdk.samtools">Cigar</A>&nbsp;cigar)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#countInsertedBases(htsjdk.samtools.SAMRecord)">countInsertedBases</A></B>(<A HREF="../../../htsjdk/samtools/SAMRecord.html" title="class in htsjdk.samtools">SAMRecord</A>&nbsp;read)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#countMismatches(htsjdk.samtools.SAMRecord, byte[])">countMismatches</A></B>(<A HREF="../../../htsjdk/samtools/SAMRecord.html" title="class in htsjdk.samtools">SAMRecord</A>&nbsp;read,
                byte[]&nbsp;referenceBases)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Calculates the number of mismatches between the read and the reference sequence provided.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#countMismatches(htsjdk.samtools.SAMRecord, byte[], boolean)">countMismatches</A></B>(<A HREF="../../../htsjdk/samtools/SAMRecord.html" title="class in htsjdk.samtools">SAMRecord</A>&nbsp;read,
                byte[]&nbsp;referenceBases,
                boolean&nbsp;bisulfiteSequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Calculates the number of mismatches between the read and the reference sequence provided.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#countMismatches(htsjdk.samtools.SAMRecord, byte[], int)">countMismatches</A></B>(<A HREF="../../../htsjdk/samtools/SAMRecord.html" title="class in htsjdk.samtools">SAMRecord</A>&nbsp;read,
                byte[]&nbsp;referenceBases,
                int&nbsp;referenceOffset)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Calculates the number of mismatches between the read and the reference sequence provided.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#countMismatches(htsjdk.samtools.SAMRecord, byte[], int, boolean)">countMismatches</A></B>(<A HREF="../../../htsjdk/samtools/SAMRecord.html" title="class in htsjdk.samtools">SAMRecord</A>&nbsp;read,
                byte[]&nbsp;referenceBases,
                int&nbsp;referenceOffset,
                boolean&nbsp;bisulfiteSequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Calculates the number of mismatches between the read and the reference sequence provided.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.List&lt;byte[]&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#generateAllKmers(int)">generateAllKmers</A></B>(int&nbsp;length)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Generates all possible unambiguous kmers (upper-case) of length and returns them as byte[]s.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#isBisulfiteConverted(byte, byte)">isBisulfiteConverted</A></B>(byte&nbsp;read,
                     byte&nbsp;reference)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#isBisulfiteConverted(byte, byte, boolean)">isBisulfiteConverted</A></B>(byte&nbsp;read,
                     byte&nbsp;reference,
                     boolean&nbsp;negativeStrand)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Checks for bisulfite conversion, C->T on the positive strand and G->A on the negative strand.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#isNoCall(byte)">isNoCall</A></B>(byte&nbsp;base)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;returns true if the value of base represents a no call</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#isValidBase(byte)">isValidBase</A></B>(byte&nbsp;b)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns true if the byte is in [acgtACGT].</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#makeCigarStringWithIndelPossibleClipping(int, int, int, int, int)">makeCigarStringWithIndelPossibleClipping</A></B>(int&nbsp;alignmentStart,
                                         int&nbsp;readLength,
                                         int&nbsp;referenceSequenceLength,
                                         int&nbsp;indelPosition,
                                         int&nbsp;indelLength)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Create a cigar string for a gapped alignment, which may have soft clipping at either end</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#makeCigarStringWithPossibleClipping(int, int, int)">makeCigarStringWithPossibleClipping</A></B>(int&nbsp;alignmentStart,
                                    int&nbsp;readLength,
                                    int&nbsp;referenceSequenceLength)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Create a simple ungapped cigar string, which might have soft clipping at either end</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;byte[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#makeReferenceFromAlignment(htsjdk.samtools.SAMRecord, boolean)">makeReferenceFromAlignment</A></B>(<A HREF="../../../htsjdk/samtools/SAMRecord.html" title="class in htsjdk.samtools">SAMRecord</A>&nbsp;rec,
                           boolean&nbsp;includeReferenceBasesForDeletions)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Produce reference bases from an aligned SAMRecord with MD string and Cigar.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#makeSoftClipCigar(int)">makeSoftClipCigar</A></B>(int&nbsp;clipLength)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#reverse(byte[], int, int)">reverse</A></B>(byte[]&nbsp;array,
        int&nbsp;offset,
        int&nbsp;len)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#reverseComplement(byte[])">reverseComplement</A></B>(byte[]&nbsp;bases)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Reverses and complements the bases in place.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#reverseComplement(byte[], int, int)">reverseComplement</A></B>(byte[]&nbsp;bases,
                  int&nbsp;offset,
                  int&nbsp;len)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#reverseComplement(java.lang.String)">reverseComplement</A></B>(java.lang.String&nbsp;sequenceData)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Calculate the reverse complement of the specified sequence
 (Stolen from Reseq)</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#reverseQualities(byte[])">reverseQualities</A></B>(byte[]&nbsp;quals)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Reverses the quals in place.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#sumQualitiesOfMismatches(htsjdk.samtools.SAMRecord, byte[])">sumQualitiesOfMismatches</A></B>(<A HREF="../../../htsjdk/samtools/SAMRecord.html" title="class in htsjdk.samtools">SAMRecord</A>&nbsp;read,
                         byte[]&nbsp;referenceBases)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Calculates the sum of qualities for mismatched bases in the read.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#sumQualitiesOfMismatches(htsjdk.samtools.SAMRecord, byte[], int)">sumQualitiesOfMismatches</A></B>(<A HREF="../../../htsjdk/samtools/SAMRecord.html" title="class in htsjdk.samtools">SAMRecord</A>&nbsp;read,
                         byte[]&nbsp;referenceBases,
                         int&nbsp;referenceOffset)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Calculates the sum of qualities for mismatched bases in the read.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#sumQualitiesOfMismatches(htsjdk.samtools.SAMRecord, byte[], int, boolean)">sumQualitiesOfMismatches</A></B>(<A HREF="../../../htsjdk/samtools/SAMRecord.html" title="class in htsjdk.samtools">SAMRecord</A>&nbsp;read,
                         byte[]&nbsp;referenceBases,
                         int&nbsp;referenceOffset,
                         boolean&nbsp;bisulfiteSequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Calculates the sum of qualities for mismatched bases in the read.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#sumQualitiesOfMismatches(htsjdk.samtools.SAMRecord, char[], int)">sumQualitiesOfMismatches</A></B>(<A HREF="../../../htsjdk/samtools/SAMRecord.html" title="class in htsjdk.samtools">SAMRecord</A>&nbsp;read,
                         char[]&nbsp;referenceBases,
                         int&nbsp;referenceOffset)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<B>Deprecated.</B>&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;byte</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#upperCase(byte)">upperCase</A></B>(byte&nbsp;base)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;byte[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../htsjdk/samtools/util/SequenceUtil.html#upperCase(byte[])">upperCase</A></B>(byte[]&nbsp;bases)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_java.lang.Object"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class java.lang.Object</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>clone, equals, finalize, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</CODE></TD>
</TR>
</TABLE>
&nbsp;
<P>

<!-- ============ FIELD DETAIL =========== -->

<A NAME="field_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Field Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="a"><!-- --></A><H3>
a</H3>
<PRE>
public static final byte <B>a</B></PRE>
<DL>
<DD>Byte typed variables for all normal bases.
<P>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../constant-values.html#htsjdk.samtools.util.SequenceUtil.a">Constant Field Values</A></DL>
</DL>
<HR>

<A NAME="c"><!-- --></A><H3>
c</H3>
<PRE>
public static final byte <B>c</B></PRE>
<DL>
<DD>Byte typed variables for all normal bases.
<P>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../constant-values.html#htsjdk.samtools.util.SequenceUtil.c">Constant Field Values</A></DL>
</DL>
<HR>

<A NAME="g"><!-- --></A><H3>
g</H3>
<PRE>
public static final byte <B>g</B></PRE>
<DL>
<DD>Byte typed variables for all normal bases.
<P>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../constant-values.html#htsjdk.samtools.util.SequenceUtil.g">Constant Field Values</A></DL>
</DL>
<HR>

<A NAME="t"><!-- --></A><H3>
t</H3>
<PRE>
public static final byte <B>t</B></PRE>
<DL>
<DD>Byte typed variables for all normal bases.
<P>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../constant-values.html#htsjdk.samtools.util.SequenceUtil.t">Constant Field Values</A></DL>
</DL>
<HR>

<A NAME="n"><!-- --></A><H3>
n</H3>
<PRE>
public static final byte <B>n</B></PRE>
<DL>
<DD>Byte typed variables for all normal bases.
<P>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../constant-values.html#htsjdk.samtools.util.SequenceUtil.n">Constant Field Values</A></DL>
</DL>
<HR>

<A NAME="A"><!-- --></A><H3>
A</H3>
<PRE>
public static final byte <B>A</B></PRE>
<DL>
<DD>Byte typed variables for all normal bases.
<P>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../constant-values.html#htsjdk.samtools.util.SequenceUtil.A">Constant Field Values</A></DL>
</DL>
<HR>

<A NAME="C"><!-- --></A><H3>
C</H3>
<PRE>
public static final byte <B>C</B></PRE>
<DL>
<DD>Byte typed variables for all normal bases.
<P>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../constant-values.html#htsjdk.samtools.util.SequenceUtil.C">Constant Field Values</A></DL>
</DL>
<HR>

<A NAME="G"><!-- --></A><H3>
G</H3>
<PRE>
public static final byte <B>G</B></PRE>
<DL>
<DD>Byte typed variables for all normal bases.
<P>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../constant-values.html#htsjdk.samtools.util.SequenceUtil.G">Constant Field Values</A></DL>
</DL>
<HR>

<A NAME="T"><!-- --></A><H3>
T</H3>
<PRE>
public static final byte <B>T</B></PRE>
<DL>
<DD>Byte typed variables for all normal bases.
<P>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../constant-values.html#htsjdk.samtools.util.SequenceUtil.T">Constant Field Values</A></DL>
</DL>
<HR>

<A NAME="N"><!-- --></A><H3>
N</H3>
<PRE>
public static final byte <B>N</B></PRE>
<DL>
<DD>Byte typed variables for all normal bases.
<P>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../constant-values.html#htsjdk.samtools.util.SequenceUtil.N">Constant Field Values</A></DL>
</DL>
<HR>

<A NAME="VALID_BASES_UPPER"><!-- --></A><H3>
VALID_BASES_UPPER</H3>
<PRE>
public static final byte[] <B>VALID_BASES_UPPER</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="VALID_BASES_LOWER"><!-- --></A><H3>
VALID_BASES_LOWER</H3>
<PRE>
public static final byte[] <B>VALID_BASES_LOWER</B></PRE>
<DL>
<DL>
</DL>
</DL>

<!-- ========= CONSTRUCTOR DETAIL ======== -->

<A NAME="constructor_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Constructor Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="SequenceUtil()"><!-- --></A><H3>
SequenceUtil</H3>
<PRE>
public <B>SequenceUtil</B>()</PRE>
<DL>
</DL>

<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Method Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="reverseComplement(java.lang.String)"><!-- --></A><H3>
reverseComplement</H3>
<PRE>
public static java.lang.String <B>reverseComplement</B>(java.lang.String&nbsp;sequenceData)</PRE>
<DL>
<DD>Calculate the reverse complement of the specified sequence
 (Stolen from Reseq)
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>sequenceData</CODE> - 
<DT><B>Returns:</B><DD>reverse complement</DL>
</DD>
</DL>
<HR>

<A NAME="basesEqual(byte, byte)"><!-- --></A><H3>
basesEqual</H3>
<PRE>
public static boolean <B>basesEqual</B>(byte&nbsp;lhs,
                                 byte&nbsp;rhs)</PRE>
<DL>
<DD>Attempts to efficiently compare two bases stored as bytes for equality.
<P>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="isNoCall(byte)"><!-- --></A><H3>
isNoCall</H3>
<PRE>
public static boolean <B>isNoCall</B>(byte&nbsp;base)</PRE>
<DL>
<DD>returns true if the value of base represents a no call
<P>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="isValidBase(byte)"><!-- --></A><H3>
isValidBase</H3>
<PRE>
public static boolean <B>isValidBase</B>(byte&nbsp;b)</PRE>
<DL>
<DD>Returns true if the byte is in [acgtACGT].
<P>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="calculateGc(byte[])"><!-- --></A><H3>
calculateGc</H3>
<PRE>
public static double <B>calculateGc</B>(byte[]&nbsp;bases)</PRE>
<DL>
<DD>Calculates the fraction of bases that are G/C in the sequence.
<P>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="assertSequenceListsEqual(java.util.List, java.util.List)"><!-- --></A><H3>
assertSequenceListsEqual</H3>
<PRE>
public static void <B>assertSequenceListsEqual</B>(java.util.List&lt;<A HREF="../../../htsjdk/samtools/SAMSequenceRecord.html" title="class in htsjdk.samtools">SAMSequenceRecord</A>&gt;&nbsp;s1,
                                            java.util.List&lt;<A HREF="../../../htsjdk/samtools/SAMSequenceRecord.html" title="class in htsjdk.samtools">SAMSequenceRecord</A>&gt;&nbsp;s2)</PRE>
<DL>
<DD>Throws an exception only if both parameters are not null
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>s1</CODE> - a list of sequence headers<DD><CODE>s2</CODE> - a second list of sequence headers</DL>
</DD>
</DL>
<HR>

<A NAME="areSequenceDictionariesEqual(htsjdk.samtools.SAMSequenceDictionary, htsjdk.samtools.SAMSequenceDictionary)"><!-- --></A><H3>
areSequenceDictionariesEqual</H3>
<PRE>
public static boolean <B>areSequenceDictionariesEqual</B>(<A HREF="../../../htsjdk/samtools/SAMSequenceDictionary.html" title="class in htsjdk.samtools">SAMSequenceDictionary</A>&nbsp;s1,
                                                   <A HREF="../../../htsjdk/samtools/SAMSequenceDictionary.html" title="class in htsjdk.samtools">SAMSequenceDictionary</A>&nbsp;s2)</PRE>
<DL>
<DD>Returns true if both parameters are null or equal, otherwise returns false
<P>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="assertSequenceDictionariesEqual(htsjdk.samtools.SAMSequenceDictionary, htsjdk.samtools.SAMSequenceDictionary)"><!-- --></A><H3>
assertSequenceDictionariesEqual</H3>
<PRE>
public static void <B>assertSequenceDictionariesEqual</B>(<A HREF="../../../htsjdk/samtools/SAMSequenceDictionary.html" title="class in htsjdk.samtools">SAMSequenceDictionary</A>&nbsp;s1,
                                                   <A HREF="../../../htsjdk/samtools/SAMSequenceDictionary.html" title="class in htsjdk.samtools">SAMSequenceDictionary</A>&nbsp;s2)</PRE>
<DL>
<DD>Throws an exception if both parameters are non-null and unequal.
<P>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="assertSequenceDictionariesEqual(htsjdk.samtools.SAMSequenceDictionary, htsjdk.samtools.SAMSequenceDictionary, java.io.File, java.io.File)"><!-- --></A><H3>
assertSequenceDictionariesEqual</H3>
<PRE>
public static void <B>assertSequenceDictionariesEqual</B>(<A HREF="../../../htsjdk/samtools/SAMSequenceDictionary.html" title="class in htsjdk.samtools">SAMSequenceDictionary</A>&nbsp;s1,
                                                   <A HREF="../../../htsjdk/samtools/SAMSequenceDictionary.html" title="class in htsjdk.samtools">SAMSequenceDictionary</A>&nbsp;s2,
                                                   java.io.File&nbsp;f1,
                                                   java.io.File&nbsp;f2)</PRE>
<DL>
<DD>Throws an exception if both parameters are non-null and unequal, including the filenames.
<P>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="makeCigarStringWithPossibleClipping(int, int, int)"><!-- --></A><H3>
makeCigarStringWithPossibleClipping</H3>
<PRE>
public static java.lang.String <B>makeCigarStringWithPossibleClipping</B>(int&nbsp;alignmentStart,
                                                                   int&nbsp;readLength,
                                                                   int&nbsp;referenceSequenceLength)</PRE>
<DL>
<DD>Create a simple ungapped cigar string, which might have soft clipping at either end
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>alignmentStart</CODE> - raw aligment start, which may result in read hanging off beginning or end of read
<DT><B>Returns:</B><DD>cigar string that may have S operator at beginning or end, and has M operator for the rest of the read</DL>
</DD>
</DL>
<HR>

<A NAME="makeCigarStringWithIndelPossibleClipping(int, int, int, int, int)"><!-- --></A><H3>
makeCigarStringWithIndelPossibleClipping</H3>
<PRE>
public static java.lang.String <B>makeCigarStringWithIndelPossibleClipping</B>(int&nbsp;alignmentStart,
                                                                        int&nbsp;readLength,
                                                                        int&nbsp;referenceSequenceLength,
                                                                        int&nbsp;indelPosition,
                                                                        int&nbsp;indelLength)</PRE>
<DL>
<DD>Create a cigar string for a gapped alignment, which may have soft clipping at either end
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>alignmentStart</CODE> - raw alignment start, which may result in read hanging off beginning or end of read<DD><CODE>readLength</CODE> - <DD><CODE>referenceSequenceLength</CODE> - <DD><CODE>indelPosition</CODE> - number of matching bases before indel.  Must be > 0<DD><CODE>indelLength</CODE> - length of indel.  Positive for insertion, negative for deletion.
<DT><B>Returns:</B><DD>cigar string that may have S operator at beginning or end, has one or two M operators, and an I or a D.</DL>
</DD>
</DL>
<HR>

<A NAME="makeSoftClipCigar(int)"><!-- --></A><H3>
makeSoftClipCigar</H3>
<PRE>
public static java.lang.String <B>makeSoftClipCigar</B>(int&nbsp;clipLength)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="countMismatches(htsjdk.samtools.SAMRecord, byte[])"><!-- --></A><H3>
countMismatches</H3>
<PRE>
public static int <B>countMismatches</B>(<A HREF="../../../htsjdk/samtools/SAMRecord.html" title="class in htsjdk.samtools">SAMRecord</A>&nbsp;read,
                                  byte[]&nbsp;referenceBases)</PRE>
<DL>
<DD>Calculates the number of mismatches between the read and the reference sequence provided.
<P>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="countMismatches(htsjdk.samtools.SAMRecord, byte[], int)"><!-- --></A><H3>
countMismatches</H3>
<PRE>
public static int <B>countMismatches</B>(<A HREF="../../../htsjdk/samtools/SAMRecord.html" title="class in htsjdk.samtools">SAMRecord</A>&nbsp;read,
                                  byte[]&nbsp;referenceBases,
                                  int&nbsp;referenceOffset)</PRE>
<DL>
<DD>Calculates the number of mismatches between the read and the reference sequence provided.
<P>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="countMismatches(htsjdk.samtools.SAMRecord, byte[], int, boolean)"><!-- --></A><H3>
countMismatches</H3>
<PRE>
public static int <B>countMismatches</B>(<A HREF="../../../htsjdk/samtools/SAMRecord.html" title="class in htsjdk.samtools">SAMRecord</A>&nbsp;read,
                                  byte[]&nbsp;referenceBases,
                                  int&nbsp;referenceOffset,
                                  boolean&nbsp;bisulfiteSequence)</PRE>
<DL>
<DD>Calculates the number of mismatches between the read and the reference sequence provided.
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>referenceBases</CODE> - Array of ASCII bytes that covers at least the the portion of the reference sequence
                          to which read is aligned from getReferenceStart to getReferenceEnd.<DD><CODE>referenceOffset</CODE> - 0-based offset of the first element of referenceBases relative to the start
                          of that reference sequence.<DD><CODE>bisulfiteSequence</CODE> - If this is true, it is assumed that the reads were bisulfite treated
                          and C->T on the positive strand and G->A on the negative strand will not be counted
                          as mismatches.</DL>
</DD>
</DL>
<HR>

<A NAME="countMismatches(htsjdk.samtools.SAMRecord, byte[], boolean)"><!-- --></A><H3>
countMismatches</H3>
<PRE>
public static int <B>countMismatches</B>(<A HREF="../../../htsjdk/samtools/SAMRecord.html" title="class in htsjdk.samtools">SAMRecord</A>&nbsp;read,
                                  byte[]&nbsp;referenceBases,
                                  boolean&nbsp;bisulfiteSequence)</PRE>
<DL>
<DD>Calculates the number of mismatches between the read and the reference sequence provided.
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>referenceBases</CODE> - Array of ASCII bytes that covers at least the the portion of the reference sequence
                          to which read is aligned from getReferenceStart to getReferenceEnd.<DD><CODE>bisulfiteSequence</CODE> - If this is true, it is assumed that the reads were bisulfite treated
                          and C->T on the positive strand and G->A on the negative strand will not be counted
                          as mismatches.</DL>
</DD>
</DL>
<HR>

<A NAME="sumQualitiesOfMismatches(htsjdk.samtools.SAMRecord, byte[])"><!-- --></A><H3>
sumQualitiesOfMismatches</H3>
<PRE>
public static int <B>sumQualitiesOfMismatches</B>(<A HREF="../../../htsjdk/samtools/SAMRecord.html" title="class in htsjdk.samtools">SAMRecord</A>&nbsp;read,
                                           byte[]&nbsp;referenceBases)</PRE>
<DL>
<DD>Calculates the sum of qualities for mismatched bases in the read.
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>referenceBases</CODE> - Array of ASCII bytes in which the 0th position in the array corresponds
                       to the first element of the reference sequence to which read is aligned.</DL>
</DD>
</DL>
<HR>

<A NAME="sumQualitiesOfMismatches(htsjdk.samtools.SAMRecord, byte[], int)"><!-- --></A><H3>
sumQualitiesOfMismatches</H3>
<PRE>
public static int <B>sumQualitiesOfMismatches</B>(<A HREF="../../../htsjdk/samtools/SAMRecord.html" title="class in htsjdk.samtools">SAMRecord</A>&nbsp;read,
                                           byte[]&nbsp;referenceBases,
                                           int&nbsp;referenceOffset)</PRE>
<DL>
<DD>Calculates the sum of qualities for mismatched bases in the read.
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>referenceBases</CODE> - Array of ASCII bytes that covers at least the the portion of the reference sequence
                        to which read is aligned from getReferenceStart to getReferenceEnd.<DD><CODE>referenceOffset</CODE> - 0-based offset of the first element of referenceBases relative to the start
                        of that reference sequence.</DL>
</DD>
</DL>
<HR>

<A NAME="sumQualitiesOfMismatches(htsjdk.samtools.SAMRecord, byte[], int, boolean)"><!-- --></A><H3>
sumQualitiesOfMismatches</H3>
<PRE>
public static int <B>sumQualitiesOfMismatches</B>(<A HREF="../../../htsjdk/samtools/SAMRecord.html" title="class in htsjdk.samtools">SAMRecord</A>&nbsp;read,
                                           byte[]&nbsp;referenceBases,
                                           int&nbsp;referenceOffset,
                                           boolean&nbsp;bisulfiteSequence)</PRE>
<DL>
<DD>Calculates the sum of qualities for mismatched bases in the read.
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>referenceBases</CODE> - Array of ASCII bytes that covers at least the the portion of the reference sequence
                          to which read is aligned from getReferenceStart to getReferenceEnd.<DD><CODE>referenceOffset</CODE> - 0-based offset of the first element of referenceBases relative to the start
                          of that reference sequence.<DD><CODE>bisulfiteSequence</CODE> - If this is true, it is assumed that the reads were bisulfite treated
                          and C->T on the positive strand and G->A on the negative strand will not be counted
                          as mismatches.</DL>
</DD>
</DL>
<HR>

<A NAME="sumQualitiesOfMismatches(htsjdk.samtools.SAMRecord, char[], int)"><!-- --></A><H3>
sumQualitiesOfMismatches</H3>
<PRE>
<FONT SIZE="-1">@Deprecated
</FONT>public static int <B>sumQualitiesOfMismatches</B>(<A HREF="../../../htsjdk/samtools/SAMRecord.html" title="class in htsjdk.samtools">SAMRecord</A>&nbsp;read,
                                                      char[]&nbsp;referenceBases,
                                                      int&nbsp;referenceOffset)</PRE>
<DL>
<DD><B>Deprecated.</B>&nbsp;
<P>
<DD>Sadly, this is a duplicate of the method above, except that it takes char[] for referenceBases rather
 than byte[].  This is because GATK needs it this way.
 <p/>
 TODO: Remove this method when GATK map method is changed to take refseq as byte[].
 TODO: UPDATE: Seems to be removed from GATK. Deprecated now to be removed in a future version.
<P>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="countInsertedBases(htsjdk.samtools.Cigar)"><!-- --></A><H3>
countInsertedBases</H3>
<PRE>
public static int <B>countInsertedBases</B>(<A HREF="../../../htsjdk/samtools/Cigar.html" title="class in htsjdk.samtools">Cigar</A>&nbsp;cigar)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="countDeletedBases(htsjdk.samtools.Cigar)"><!-- --></A><H3>
countDeletedBases</H3>
<PRE>
public static int <B>countDeletedBases</B>(<A HREF="../../../htsjdk/samtools/Cigar.html" title="class in htsjdk.samtools">Cigar</A>&nbsp;cigar)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="countInsertedBases(htsjdk.samtools.SAMRecord)"><!-- --></A><H3>
countInsertedBases</H3>
<PRE>
public static int <B>countInsertedBases</B>(<A HREF="../../../htsjdk/samtools/SAMRecord.html" title="class in htsjdk.samtools">SAMRecord</A>&nbsp;read)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="countDeletedBases(htsjdk.samtools.SAMRecord)"><!-- --></A><H3>
countDeletedBases</H3>
<PRE>
public static int <B>countDeletedBases</B>(<A HREF="../../../htsjdk/samtools/SAMRecord.html" title="class in htsjdk.samtools">SAMRecord</A>&nbsp;read)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="calculateSamNmTag(htsjdk.samtools.SAMRecord, byte[])"><!-- --></A><H3>
calculateSamNmTag</H3>
<PRE>
public static int <B>calculateSamNmTag</B>(<A HREF="../../../htsjdk/samtools/SAMRecord.html" title="class in htsjdk.samtools">SAMRecord</A>&nbsp;read,
                                    byte[]&nbsp;referenceBases)</PRE>
<DL>
<DD>Calculates the for the predefined NM tag from the SAM spec. To the result of
 countMismatches() it adds 1 for each indel.
<P>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="calculateSamNmTag(htsjdk.samtools.SAMRecord, byte[], int)"><!-- --></A><H3>
calculateSamNmTag</H3>
<PRE>
public static int <B>calculateSamNmTag</B>(<A HREF="../../../htsjdk/samtools/SAMRecord.html" title="class in htsjdk.samtools">SAMRecord</A>&nbsp;read,
                                    byte[]&nbsp;referenceBases,
                                    int&nbsp;referenceOffset)</PRE>
<DL>
<DD>Calculates the for the predefined NM tag from the SAM spec. To the result of
 countMismatches() it adds 1 for each indel.
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>referenceOffset</CODE> - 0-based offset of the first element of referenceBases relative to the start
                        of that reference sequence.</DL>
</DD>
</DL>
<HR>

<A NAME="calculateSamNmTag(htsjdk.samtools.SAMRecord, byte[], int, boolean)"><!-- --></A><H3>
calculateSamNmTag</H3>
<PRE>
public static int <B>calculateSamNmTag</B>(<A HREF="../../../htsjdk/samtools/SAMRecord.html" title="class in htsjdk.samtools">SAMRecord</A>&nbsp;read,
                                    byte[]&nbsp;referenceBases,
                                    int&nbsp;referenceOffset,
                                    boolean&nbsp;bisulfiteSequence)</PRE>
<DL>
<DD>Calculates the for the predefined NM tag from the SAM spec. To the result of
 countMismatches() it adds 1 for each indel.
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>referenceOffset</CODE> - 0-based offset of the first element of referenceBases relative to the start
                          of that reference sequence.<DD><CODE>bisulfiteSequence</CODE> - If this is true, it is assumed that the reads were bisulfite treated
                          and C->T on the positive strand and G->A on the negative strand will not be counted
                          as mismatches.</DL>
</DD>
</DL>
<HR>

<A NAME="calculateSamNmTagFromCigar(htsjdk.samtools.SAMRecord)"><!-- --></A><H3>
calculateSamNmTagFromCigar</H3>
<PRE>
public static int <B>calculateSamNmTagFromCigar</B>(<A HREF="../../../htsjdk/samtools/SAMRecord.html" title="class in htsjdk.samtools">SAMRecord</A>&nbsp;record)</PRE>
<DL>
<DD>Attempts to calculate the for the predefined NM tag from the SAM spec using the cigar string alone.
 It may calculate incorrectly if ambiguous operators (Like M) are used.

 Needed for testing infrastructure: SAMRecordSetBuilder
<P>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="calculateSamNmTag(htsjdk.samtools.SAMRecord, char[], int)"><!-- --></A><H3>
calculateSamNmTag</H3>
<PRE>
<FONT SIZE="-1">@Deprecated
</FONT>public static int <B>calculateSamNmTag</B>(<A HREF="../../../htsjdk/samtools/SAMRecord.html" title="class in htsjdk.samtools">SAMRecord</A>&nbsp;read,
                                               char[]&nbsp;referenceBases,
                                               int&nbsp;referenceOffset)</PRE>
<DL>
<DD><B>Deprecated.</B>&nbsp;
<P>
<DD>Sadly, this is a duplicate of the method above, except that it takes char[] for referenceBases rather
 than byte[].  This is because GATK needs it this way.
 <p/>
 TODO: Remove this method when GATK map method is changed to take refseq as byte[].
 TODO: UPDATE: Seems to be removed from GATK. Deprecated now to be removed in a future version.
<P>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="complement(byte)"><!-- --></A><H3>
complement</H3>
<PRE>
public static byte <B>complement</B>(byte&nbsp;b)</PRE>
<DL>
<DD>Returns the complement of a single byte.
<P>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="reverseComplement(byte[])"><!-- --></A><H3>
reverseComplement</H3>
<PRE>
public static void <B>reverseComplement</B>(byte[]&nbsp;bases)</PRE>
<DL>
<DD>Reverses and complements the bases in place.
<P>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="reverseQualities(byte[])"><!-- --></A><H3>
reverseQualities</H3>
<PRE>
public static void <B>reverseQualities</B>(byte[]&nbsp;quals)</PRE>
<DL>
<DD>Reverses the quals in place.
<P>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="bisulfiteBasesEqual(boolean, byte, byte)"><!-- --></A><H3>
bisulfiteBasesEqual</H3>
<PRE>
public static boolean <B>bisulfiteBasesEqual</B>(boolean&nbsp;negativeStrand,
                                          byte&nbsp;read,
                                          byte&nbsp;reference)</PRE>
<DL>
<DD>Returns true if the bases are equal OR if the mismatch cannot be accounted for by
 bisfulite treatment.  C->T on the positive strand and G->A on the negative strand
 do not count as mismatches
<P>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="bisulfiteBasesEqual(byte, byte)"><!-- --></A><H3>
bisulfiteBasesEqual</H3>
<PRE>
public static boolean <B>bisulfiteBasesEqual</B>(byte&nbsp;read,
                                          byte&nbsp;reference)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="isBisulfiteConverted(byte, byte, boolean)"><!-- --></A><H3>
isBisulfiteConverted</H3>
<PRE>
public static boolean <B>isBisulfiteConverted</B>(byte&nbsp;read,
                                           byte&nbsp;reference,
                                           boolean&nbsp;negativeStrand)</PRE>
<DL>
<DD>Checks for bisulfite conversion, C->T on the positive strand and G->A on the negative strand.
<P>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="isBisulfiteConverted(byte, byte)"><!-- --></A><H3>
isBisulfiteConverted</H3>
<PRE>
public static boolean <B>isBisulfiteConverted</B>(byte&nbsp;read,
                                           byte&nbsp;reference)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="makeReferenceFromAlignment(htsjdk.samtools.SAMRecord, boolean)"><!-- --></A><H3>
makeReferenceFromAlignment</H3>
<PRE>
public static byte[] <B>makeReferenceFromAlignment</B>(<A HREF="../../../htsjdk/samtools/SAMRecord.html" title="class in htsjdk.samtools">SAMRecord</A>&nbsp;rec,
                                                boolean&nbsp;includeReferenceBasesForDeletions)</PRE>
<DL>
<DD>Produce reference bases from an aligned SAMRecord with MD string and Cigar.
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>rec</CODE> - Must contain non-empty CIGAR and MD attribute.<DD><CODE>includeReferenceBasesForDeletions</CODE> - If true, include reference bases that are deleted in the read.
                                          This will make the returned array not line up with the read if there are deletions.
<DT><B>Returns:</B><DD>References bases corresponding to the read.  If there is an insertion in the read, reference contains
 '-'.  If the read is soft-clipped, reference contains '0'.  If there is a skipped region and
 includeReferenceBasesForDeletions==true, reference will have Ns for the skipped region.</DL>
</DD>
</DL>
<HR>

<A NAME="reverse(byte[], int, int)"><!-- --></A><H3>
reverse</H3>
<PRE>
public static void <B>reverse</B>(byte[]&nbsp;array,
                           int&nbsp;offset,
                           int&nbsp;len)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="reverseComplement(byte[], int, int)"><!-- --></A><H3>
reverseComplement</H3>
<PRE>
public static void <B>reverseComplement</B>(byte[]&nbsp;bases,
                                     int&nbsp;offset,
                                     int&nbsp;len)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="calculateMD5String(byte[])"><!-- --></A><H3>
calculateMD5String</H3>
<PRE>
public static java.lang.String <B>calculateMD5String</B>(byte[]&nbsp;data)
                                           throws java.security.NoSuchAlgorithmException</PRE>
<DL>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE>java.security.NoSuchAlgorithmException</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="calculateMD5String(byte[], int, int)"><!-- --></A><H3>
calculateMD5String</H3>
<PRE>
public static java.lang.String <B>calculateMD5String</B>(byte[]&nbsp;data,
                                                  int&nbsp;offset,
                                                  int&nbsp;len)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="calculateMD5(byte[], int, int)"><!-- --></A><H3>
calculateMD5</H3>
<PRE>
public static byte[] <B>calculateMD5</B>(byte[]&nbsp;data,
                                  int&nbsp;offset,
                                  int&nbsp;len)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="calculateMdAndNmTags(htsjdk.samtools.SAMRecord, byte[], boolean, boolean)"><!-- --></A><H3>
calculateMdAndNmTags</H3>
<PRE>
public static void <B>calculateMdAndNmTags</B>(<A HREF="../../../htsjdk/samtools/SAMRecord.html" title="class in htsjdk.samtools">SAMRecord</A>&nbsp;record,
                                        byte[]&nbsp;ref,
                                        boolean&nbsp;calcMD,
                                        boolean&nbsp;calcNM)</PRE>
<DL>
<DD>A rip off samtools bam_md.c
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>record</CODE> - <DD><CODE>ref</CODE> - <DD><CODE>calcMD</CODE> - </DL>
</DD>
</DL>
<HR>

<A NAME="upperCase(byte)"><!-- --></A><H3>
upperCase</H3>
<PRE>
public static byte <B>upperCase</B>(byte&nbsp;base)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="upperCase(byte[])"><!-- --></A><H3>
upperCase</H3>
<PRE>
public static byte[] <B>upperCase</B>(byte[]&nbsp;bases)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="generateAllKmers(int)"><!-- --></A><H3>
generateAllKmers</H3>
<PRE>
public static java.util.List&lt;byte[]&gt; <B>generateAllKmers</B>(int&nbsp;length)</PRE>
<DL>
<DD>Generates all possible unambiguous kmers (upper-case) of length and returns them as byte[]s.
<P>
<DD><DL>
</DL>
</DD>
</DL>
<!-- ========= END OF CLASS DATA ========= -->
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/SequenceUtil.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../index-all.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../htsjdk/samtools/util/SamRecordWithOrdinal.html" title="class in htsjdk.samtools.util"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../htsjdk/samtools/util/SequenceUtil.SequenceListsDifferException.html" title="class in htsjdk.samtools.util"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../index.html?htsjdk/samtools/util/SequenceUtil.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="SequenceUtil.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;<A HREF="#nested_class_summary">NESTED</A>&nbsp;|&nbsp;<A HREF="#field_summary">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;<A HREF="#field_detail">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
